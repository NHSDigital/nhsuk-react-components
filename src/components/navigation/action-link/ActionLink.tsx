import React, { FC } from 'react';
import classNames from 'classnames';
import { ArrowRightCircle } from '@components/content-presentation/icons';
import { AsElementLink } from '@util/types/LinkTypes';

const ActionLink: FC<AsElementLink<HTMLAnchorElement>> = ({
  children,
  asElement: Component = 'a',
  className,
  ...rest
}) => (
  <div className="nhsuk-action-link">
    <Component className={classNames('nhsuk-action-link__link', className)} {...rest}>
      <ArrowRightCircle width={36} height={36} />
      <span className="nhsuk-action-link__text">{children}</span>
    </Component>
  </div>
);

export default ActionLink;
