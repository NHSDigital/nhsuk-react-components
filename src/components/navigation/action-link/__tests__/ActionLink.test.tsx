import React from 'react';
import { render } from '@testing-library/react';
import ActionLink from '../ActionLink';

describe('ActionLink', () => {
  it('matches snapshot', () => {
    const { container } = render(<ActionLink href="/test">Test</ActionLink>);

    expect(container).toMatchSnapshot('ActionLink');
  });
});
