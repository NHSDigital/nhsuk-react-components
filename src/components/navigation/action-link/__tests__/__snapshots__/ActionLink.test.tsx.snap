// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionLink matches snapshot: ActionLink 1`] = `
<div>
  <div
    class="nhsuk-action-link"
  >
    <a
      class="nhsuk-action-link__link"
      href="/test"
    >
      <svg
        aria-hidden="true"
        class="nhsuk-icon nhsuk-icon__arrow-right-circle"
        height="36"
        viewBox="0 0 24 24"
        width="36"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z"
        />
      </svg>
      <span
        class="nhsuk-action-link__text"
      >
        Test
      </span>
    </a>
  </div>
</div>
`;
